neqo-crypto/src/aead.rs

```diff
@@ -126,6 +126,39 @@ impl RealAead {
         Ok(&output[0..(l.try_into()?)])
     }
 
+    /// Encrypt `data` consisting of `aad` and plaintext `input` in place.
+    ///
+    /// The space provided in `data` needs to allow `Aead::expansion` more bytes to be appended.
+    ///
+    /// # Errors
```

Comment:
This needs panics docs.  You are taking a range argument and indexing a slice argument.  That's going to break.